### ISSUE TITLE
Error in `Layout/SpaceInsideArrayLiteralBrackets` when pattern specifies a class

### ISSUE THREAD

**Issue by a-bulancov** [rubocop/rubocop#14160]
Rubocop 1.75.5 improved `Layout/SpaceInsideArrayLiteralBrackets` to also check pattern matching arrays (#14144). However, this fails if a class is specified in the pattern.

`Layout/SpaceInsideHashLiteralBraces` does not seem to have this problem, possibly because it expects curly braces, which are incompatible with class specification.

--------

## Expected behavior

I expected the class to get ignored.

## Actual behavior

Cop raises an error:
```
$ rubocop -d example.rb
For /Users/alex: Default configuration from /Users/alex/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.75.5/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/alex/example.rb
An error occurred while Layout/SpaceInsideArrayLiteralBrackets cop was inspecting /Users/alex/example.rb:7:9.
undefined method `+' for nil
/Users/alex/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.75.5/lib/rubocop/cop/layout/space_inside_array_literal_brackets.rb:132:in `next_to_newline?'
/Users/alex/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.75.5/lib/rubocop/cop/layout/space_inside_array_literal_brackets.rb:95:in `on_array'
/Users/alex/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.75.5/lib/rubocop/cop/commissioner.rb:107:in `public_send'
/Users/alex/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.75.5/lib/rubocop/cop/commissioner.rb:107:in `block (2 levels) in trigger_responding_cops'
/Users/alex/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.75.5/lib/rubocop/cop/commissioner.rb:171:in `with_cop_error_handling'
/Users/alex/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.75.5/lib/rubocop/cop/commissioner.rb:106:in `block in trigger_responding_cops'
/Users/alex/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.75.5/lib/rubocop/cop/commissioner.rb:105:in `each'
/Users/alex/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.75.5/lib/rubocop/cop/commissioner.rb:105:in `trigger_responding_cops'
/Users/alex/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-1.75.5/lib/rubocop/cop/commissioner.rb:69:in `on_array_pattern'
/Users/alex/.rbenv/versions/3.3.5/lib/ruby/gems/3.3.0/gems/rubocop-ast-1.44.1/lib/rubocop/ast/traversal.rb:146:in `block in on_dstr'
# ... omitting the rest of the backtrace, it's just dispatch
```

## Steps to reproduce the problem

This file exhibits the error on line 7 (`in Array[1,2,3]`), though the exact class does not matter:
```ruby
# frozen_string_literal: true

a = [1, 2, 3]
case a
in [1, 2]
  puts '1,2'
in Array[1, 2, 3]
  puts '1,2,3'
in Hash[a: 1]
  puts 'Hash!'
else
  puts 'who knows?'
end
```

## RuboCop version

```
$ rubocop -V
1.75.5 (using Parser 3.3.8.0, rubocop-ast 1.44.1, analyzing as Ruby 2.7, running on ruby 3.3.5) [x86_64-darwin23]
```
(but the same happens when analyzing as Ruby 3.3)

**Comment by a-bulancov** [comment]
I should also mention that the error is the same if round brackets are used (`in Array(1, 2, 3)`).

**Comment by a-bulancov** [comment]
Sorry for a duplicate, and thank you!
